kotlin/display-grid/src/main/java/com/esri/arcgisruntime/sample/displaygrid/MainActivity.kt

```diff
@@ -0,0 +1,299 @@
+          this@MainActivity, android.R.layout.simple_spinner_item,
+          resources.getStringArray(R.array.colors_array)
+        ).also { labelColorAdapter ->
+          labelColorAdapter.setDropDownViewResource(android.R.layout.simple_spinner_dropdown_item)
+        }
+
+        // change grid labels color
+        onItemSelectedListener = object : OnItemSelectedListener {
+          override fun onItemSelected(
+            parent: AdapterView<*>?,
+            view: View,
+            position: Int,
+            id: Long
+          ) {
+            // set the color
+            when (position) {
+              0 -> labelColor = Color.RED
+              1 -> labelColor = Color.WHITE
+              2 -> labelColor = Color.BLUE
+              else -> Toast.makeText(
+                this@MainActivity,
+                "Unsupported option",
+                Toast.LENGTH_SHORT
+              ).show()
+            }
+            changeLabelColor(labelColor)
+          }
+
+          override fun onNothingSelected(parent: AdapterView<*>?) {}
+        }
+      }
+
+      // create drop-down list of different label positions
+      label_position_spinner.apply {
+        adapter = ArrayAdapter(
+          this@MainActivity, android.R.layout.simple_spinner_item,
+          resources.getStringArray(R.array.positions_array)
+        ).also { positionAdapter ->
+          positionAdapter.setDropDownViewResource(android.R.layout.simple_spinner_dropdown_item)
+        }
+
+        onItemSelectedListener = object : OnItemSelectedListener {
+          override fun onItemSelected(
+            parent: AdapterView<*>?,
+            view: View,
+            position: Int,
+            id: Long
+          ) {
+            // set the label position
+            when (position) {
+              0 -> labelPosition = Grid.LabelPosition.ALL_SIDES
+              1 -> labelPosition = Grid.LabelPosition.BOTTOM_LEFT
+              2 -> labelPosition = Grid.LabelPosition.BOTTOM_RIGHT
+              3 -> labelPosition = Grid.LabelPosition.CENTER
+              4 -> labelPosition = Grid.LabelPosition.GEOGRAPHIC
+              5 -> labelPosition = Grid.LabelPosition.TOP_LEFT
+              6 -> labelPosition = Grid.LabelPosition.TOP_RIGHT
+              else -> Toast.makeText(
+                this@MainActivity,
+                "Unsupported option",
+                Toast.LENGTH_SHORT
+              ).show()
+            }
+            changeLabelPosition(labelPosition)
+          }
+
+          override fun onNothingSelected(parent: AdapterView<*>?) {}
+        }
+      }
+
+      labels_checkBox.apply {
+        isChecked = true
+        // hide and show label visibility when the checkbox is clicked
+        setOnClickListener {
+          setLabelVisibility(this.isChecked)
+        }
+      }
+    }
+
+    // display pop-up box when button is clicked
+    menu_button.setOnClickListener { dialog.show() }
+  }
+  
+  private fun setLabelVisibility(visible: Boolean) {
```

Comment:
don't forget to add KDoc for all of these functions :)

Reply:
Thanks!